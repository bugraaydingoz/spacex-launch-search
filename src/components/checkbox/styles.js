import styled from 'styled-components'

export const Container = styled.label`
  display: flex;
  position: relative;
  padding-left: 30px;
  margin-bottom: 20px;
  background-color: rgba(0, 0, 0, 0);
  cursor: pointer;
  font-size: 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
`

export const Input = styled.input`
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
`

export const Checkmark = styled.span`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background-color: rgba(0, 0, 0, 0);
  border: var(--border);

  &:after {
    display: ${({ checked }) => (checked ? 'block' : 'none')};
    position: absolute;
    content: '';
    left: 5px;
    top: 1px;
    width: 3px;
    height: 8px;
    border: solid white;
    border-width: 0 3px 3px 0;
    -webkit-transform: rotate(45deg);
    -ms-transform: rotate(45deg);
    transform: rotate(45deg);
  }
`
