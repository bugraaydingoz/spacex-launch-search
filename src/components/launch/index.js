export { Launch } from './Launch'
export { LaunchList } from './LaunchList'
export { LaunchLoading } from './LaunchLoading'
