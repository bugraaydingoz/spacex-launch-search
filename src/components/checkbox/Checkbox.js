import React from 'react'

// Styles
import { Container, Input, Checkmark } from './styles'

export function Checkbox({ checked, label, onChange }) {
  return (
    <Container>
      {label}
      <Input
        type="checkbox"
        onChange={() => onChange(!checked)}
        checked={checked}
      />
      <Checkmark checked={checked} />
    </Container>
  )
}
