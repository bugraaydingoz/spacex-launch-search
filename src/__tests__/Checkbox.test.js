import React from 'react'
import renderer from 'react-test-renderer'
import { Checkbox } from '../components/checkbox'

test('renders correctly', () => {
  const tree = renderer
    .create(<Checkbox checked={true} label="Test" onChange={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
