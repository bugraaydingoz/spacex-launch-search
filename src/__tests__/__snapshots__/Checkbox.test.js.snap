// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className="sc-AxjAm jtKjGz"
>
  Test
  <input
    checked={true}
    className="sc-AxirZ cEJisw"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    checked={true}
    className="sc-AxiKw lgQkCS"
  />
</label>
`;
